sdk/lib/locks/locks.cc

```diff
@@ -136,14 +136,33 @@ namespace
 				lockWord.notify_all();
 			}
 		}
+
+		/**
+		 * Set the destruction bit in the flag lock word and wake
+		 * waiters. Assumes that the lock is held by the caller.
+		 *
+		 * Note: This does not check that the lock is owned by the
```

Comment:
It might be nice to have an assert that does check.  There's a `Debug::Assert` variant that takes a lambda as the argument and so will not compile the check unless we've turned on lock debugging.

Reply:
I agree, these would be nice to have!

I suggest to add them in a separate commit. Currently, most methods in this file make this assumption without any assert. The commit which adds the assert could do so on all methods at once.